reconcile/change_owners/change_log_tracking.py

```diff
@@ -101,6 +104,31 @@ def run(
             changes = aggregate_file_moves(parse_bundle_changes(diff))
             for change in changes:
                 logging.debug(f"Processing change {change}")
+                match change.fileref.schema:
+                    case "/app-sre/app-1.yml":
+                        old_app_name = change.old["name"] if change.old else None
+                        if old_app_name:
+                            change_log_item.apps.append(old_app_name)
+                        new_app_name = change.new["name"] if change.new else None
+                        if new_app_name:
+                            change_log_item.apps.append(new_app_name)
+                    case (
+                        "/app-sre/saas-file-2.yml"
+                        | "/openshift/namespace-1.yml"
+                        | "/dependencies/jenkins-config-1.yml"
+                    ):
+                        old_app_path = change.old["app"]["$ref"] if change.old else None
+                        new_app_path = change.new["app"]["$ref"] if change.new else None
+                        changed_apps = {
+                            a.name
+                            for a in apps
+                            if a.path in {old_app_path, new_app_path}
+                        }
```

Comment:
optimize lookup time by build a dict `app_name_by_path` of `path -> name` beforehand, then this part can be

```python
    app_name_by_path = {a.path: a.name for a in apps}
```

```python
    diffs = filter([change.old, change.new])
    changed_apps = {
        name
        for c in diffs
        if (name := app_name_by_path.get(change["app"]["$ref"])
    }
```

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4638/commits/af2efb53e0f68f0df594fcdda95a04c8e4574317